packages/server/modules/core/tests/integration/create-user.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { expect } from 'chai'
+import { createUser, getUser, getUserById } from '@/modules/core/services/users'
+import { beforeEach, describe, it } from 'mocha'
+import { beforeEachContext } from '@/test/hooks'
+import { UserRecord } from '@/modules/core/helpers/types'
+import knexInstance from '@/db/knex'
+import {
+  createRandomEmail,
+  createRandomPassword
+} from '@/modules/core/helpers/testHelpers'
+import { USER_EMAILS_TABLE_NAME } from '@/modules/core/dbSchema'
+import { expectToThrow } from '@/test/assertionHelper'
+import { PasswordTooShortError } from '../../errors/userinput'
+
+const db = knexInstance
+const userEmailsDB = db(USER_EMAILS_TABLE_NAME)
+
+describe('Users @core-users', () => {
+  beforeEach(async () => {
+    await beforeEachContext()
+  })
+
+  it('Should create a user', async () => {
+    const newUser = {
+      name: 'John Doe',
+      email: createRandomEmail(),
+      password: createRandomPassword()
+    }
+
+    const actorId = await createUser(newUser)
+
+    expect(actorId).to.be.a('string')
+  })
+
+  it('Should store user email lowercase', async () => {
+    const user = {
+      name: 'Marty McFly',
+      email: createRandomEmail(),
+      password: createRandomPassword()
+    }
+
+    const userId = await createUser(user)
+
+    const storedUser = await getUser(userId)
+    expect(storedUser.email).to.equal(user.email.toLowerCase())
+  })
+  it('Should not create a user with a too small password', async () => {
+    const err = (await expectToThrow(() =>
+      createUser({
+        name: 'Dim Sum',
+        email: createRandomEmail(),
+        password: createRandomPassword(5)
+      })
+    )) as Error
```

Comment:
The `X as Error` type cast is unnecessary here and in other tests as well. The only other thing it can be is a `UnexpectedErrorStructureError` which is already a subclass of `Error` and thus has all of the same properties.

So there's no benefit to it + you also break TypeScript's understanding of what the value is, because it **can** in theory also be a `UnexpectedErrorStructureError` if the test breaks, even if its very unlikely